### ISSUE TITLE
bug: GraphQL context not propagated to data loaders

### ISSUE THREAD

**Issue by njuro** [Netflix/dgs-framework#2058]
Hi, we are trying to migrate to the new DGS Spring Boot starter (`graphql-dgs-spring-graphql-starter`). One problem we cannot overcome is this - we have `GraphQLContextContributor` storing the value into GraphQL context for further usage like this (simplified):

```kotlin
@Component
class EntityContextContributor: GraphQLContextContributor {
    override fun contribute(
        builder: GraphQLContext.Builder,
        extensions: Map<String, Any>?,
        requestData: DgsRequestData?,
    ) {
        builder.put("foo", "some-value")
    }
}
```
Next we have batch data loader defined like:
```kotlin
    @DgsDataLoader(name = "entities")
    class BatchEntitiesDataLoader(private val dataLoaderExecutor: Executor) : MappedBatchLoaderWithContext<String, String> {
        override fun load(
            keys: Set<String>,
            environment: BatchLoaderEnvironment,
        ): CompletionStage<Map<String, String>> {
            return CompletableFuture.supplyAsync(
                {
                    environment.getContext<GraphQLContext>.get<String>("foo")
                },
                dataLoaderExecutor,
            )
        }
    }

```

Which is invoked via 
```kotlin
    @DgsData(parentType = "Entity", field = "someField")
    fun entitySomeField(dfe: DgsDataFetchingEnvironment): CompletableFuture<String> {
        val loader = dfe.getDataLoader<String, String>(BatchEntitiesDataLoader::class.java)
        return loader.load(dfe.getSource<Entity>().id)
    }
```

Until now this has worked nicely and we could access the stored context value for key `foo` in the batch loader, but after switching to the new starter, the loader environment has now an different, empty GraphQL context which doesn't contain the value stored in the context contributor (we made sure the contributor is invoked and actually stores the value to the context it is accessing).

## Expected behavior
The GraphQL context is properly propagated to the `BatchLoaderEnvironment` and we can access the value stored in context contributor.
## Actual behavior
The `BatchLoaderEnvironment` has different, empty GraphQL context.
## Steps to reproduce
See above

**Is this a bug, or are there some changes required after switching to the new starter?**

**Comment by paulbakker** [comment]
@njuro Just confirming that we've identified the issue and are working towards a fix. We might need to make changes in Spring GraphQL, so the fix might take a little longer.

**Comment by paulbakker** [comment]
Fixed by #2067
